GraknConsole.java

```diff
@@ -210,45 +220,62 @@ private void runQuery(GraknClient.Transaction tx, String queryString) {
                 Numeric answer = tx.query().match(query.asMatchAggregate()).get();
                 printer.numeric(answer);
             } else if (query instanceof GraqlMatch.Group) {
                 Stream<ConceptMapGroup> result = tx.query().match(query.asMatchGroup());
-                printCancellableResult(result, x -> printer.conceptMapGroup(x, tx));
+                printCancellableResult(executorService, result, x -> printer.conceptMapGroup(x, tx));
             } else if (query instanceof GraqlMatch.Group.Aggregate) {
                 Stream<NumericGroup> result = tx.query().match(query.asMatchGroupAggregate());
-                printCancellableResult(result, x -> printer.numericGroup(x, tx));
+                printCancellableResult(executorService, result, x -> printer.numericGroup(x, tx));
             } else if (query instanceof GraqlCompute) {
                 throw new GraknClientException("Compute query is not yet supported");
             }
         }
+
+        executorService.shutdown();
+        try {
+            if (!executorService.awaitTermination(800, TimeUnit.MILLISECONDS)) {
+                executorService.shutdownNow();
+            }
+        } catch (InterruptedException e) {
+            executorService.shutdownNow();
+        }
     }
 
-    private <T> void printCancellableResult(Stream<T> results, Consumer<T> printFn) {
-        long counter = 0;
+    private <T> void printCancellableResult(ExecutorService executorService, Stream<T> results, Consumer<T> printFn) {
+        AtomicLong counter = new AtomicLong();
         Instant start = Instant.now();
-
         try {
-            boolean[] isCancelled = new boolean[1];
-            terminal.handle(Terminal.Signal.INT, s -> isCancelled[0] = true);
             Iterator<T> iterator = results.iterator();
-            while (!isCancelled[0] && iterator.hasNext()) {
-                counter++;
-                printFn.accept(iterator.next());
-            }
+            Future<?> answerPrintingJob = executorService.submit(() -> {
+                while (iterator.hasNext()) {
+                    printFn.accept(iterator.next());
+                    counter.getAndIncrement();
+                }
+            });
+            terminal.handle(Terminal.Signal.INT, s -> answerPrintingJob.cancel(true));
+            answerPrintingJob.get();
+            Instant end = Instant.now();
+            printer.info("answers: " + counter + ", duration: " + Duration.between(start, end).toMillis() +" ms");
+        } catch (InterruptedException | ExecutionException e) {
+            e.printStackTrace();
+        } catch (CancellationException e) {
+            Instant end = Instant.now();
+            printer.info("answers: " + counter + ", duration: " + Duration.between(start, end).toMillis() +" ms");
+
+            printer.info("Stream was shut down in the console, but the Grakn Server might still be working on last answers for some time.");
```

Comment:
canceled stream still needs to print the number of answers and duration. Also, added a message to indicate to the user that the server might be busy for some time